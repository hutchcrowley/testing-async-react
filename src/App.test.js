import React from "react";
// import renderer from "react-test-renderer";
import * as rtl from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import App from "./App";

// jest.useFakeTimers();
afterEach(rtl.cleanup);

it("Renders the heading", () => {
  // Render the app into an in-memory DOM so that we can test against it
  const wrapper = rtl.render(<span className="header">Hey Hey Hey</span>);

  // The element is now rendered in the virtual DOM, so we must get it with a query
  // The regexp /Testing React Components/i simply returns the string irrespective of case (upper or lower)

  //  test will fail if element isn't 'visible' (rendered)
});

test("<App /> snapshot", () => {
  const wrapper = rtl.render(<App />);

  expect(wrapper.asFragment()).toMatchSnapshot();
});
