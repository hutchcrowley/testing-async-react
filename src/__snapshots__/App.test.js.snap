// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <h1>
        Testing React Components
      </h1>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </header>
  </div>
</DocumentFragment>
`;
